Refactor/ec 320 regularsync

agnikowa opened the pull request:
Transform all vars from `RegularSync` class into params

Commit:
Transform first 3 vars into params (waitingForActor, headersQueue and topOfTheChain)

Commit:
Transform second 3 vars into params (resolvingBranches, resumeRegularSyncTimeout and missingStateNodeRetry)

Commit:
Remove Thread.sleep and use eventually block instead

Commit:
Fix formatting

Commit:
Resolve conflicts after merge with branch release2_0

Commit:
Merge branch 'phase/release2_0' into refactor/ec-320_regularsync

Commit:
Merge branch 'phase/release2_0' into refactor/ec-320_regularsync

Commit:
Merge branch 'phase/release2_0' into refactor/ec-320_regularsync

Commit:
Fix formatting

Commit:
Improve formatting

Commit:
Merge branch 'phase/release2_0' into refactor/ec-320_regularsync

Commit:
Replace Thread.sleep with time.advance

mmrozek commented on src/main/scala/io/iohk/ethereum/blockchain/sync/RegularSync.scala:
```diff
@@ -12,166 +12,233 @@ import io.iohk.ethereum.network.EtcPeerManagerActor.PeerInfo
 
-  def handleResumingAndPrinting: Receive = {
+  /**
+    * @param waitingForActor when on top of the chain and handling newBlockHashes message
```
Where is the rest of params?

mmrozek commented on src/main/scala/io/iohk/ethereum/blockchain/sync/RegularSync.scala:
```diff
@@ -12,166 +12,233 @@ import io.iohk.ethereum.network.EtcPeerManagerActor.PeerInfo
+    *                        dependent on our current best block info (stored in this actor to avoid DB lookups).
+    */
+  def running(
+    waitingForActor: Option[ActorRef],
```
In many methods you are using the same list of parameters. Maybe it will be more readable if you extract this fields to some State case class

mmrozek commented on src/main/scala/io/iohk/ethereum/blockchain/sync/RegularSync.scala:
```diff
@@ -12,166 +12,233 @@ import io.iohk.ethereum.network.EtcPeerManagerActor.PeerInfo
-              blacklist(peerId, blacklistDuration, error)
-          }
+          case Success(result) =>
+            val headerHash = hash2string(newBlock.header.hash)
```
`headerHash` and `newNumber` could be lazy because not all cases are using both of them

mmrozek commented on src/test/scala/io/iohk/ethereum/blockchain/sync/RegularSyncSpec.scala:
```diff
@@ -373,32 +372,31 @@ class RegularSyncSpec extends TestKit(ActorSystem("RegularSync_system")) with Wo
-    override implicit lazy val system: ActorSystem = testkitActorSystem
+    override implicit lazy val system: ActorSystem = testKitActorSystem
+
+    val time = new VirtualTime
```
`time.advance` won't be working, because you don't pass the `time` to the actor. You are passing `system.scheduler` instead of `time.scheduler`

KonradStaniec reviewed (commented):
Code looks good (much better than before) but when i tried to sync on this branch i have got: 


`12:32:41 [i.i.e.blockchain.sync.RegularSync] - Requesting 200 headers, starting from 1
12:32:41 [i.i.e.blockchain.sync.RegularSync] - Received 200 block headers in 189 ms from Peer(/94.130.140.225:30304,Actor[akka://mantis_system/user/peer-manager/94.130.140.225:30304#1611959865],false) (branch resolution: false)
12:32:42 [i.i.e.blockchain.sync.RegularSync] - Received 128 block bodies in 61 ms
12:32:45 [i.i.e.blockchain.sync.RegularSync] - Got new blocks up till block: 1 with hash 88e96d4537bea4d9c05d12549907b32561d3bf31f45aae734cdc119f13406cb6
12:32:45 [i.i.e.blockchain.sync.RegularSync] - Received 128 block bodies in 55 ms
12:32:45 [i.i.e.blockchain.sync.RegularSync] - Got new blocks up till block: 129 with hash 5656b852baa80ce4db00c60998f5cf6e7a8d76f0339d3cf97955d933f731fecf
12:32:45 [i.i.e.blockchain.sync.RegularSync] - Blacklisting peer (PeerId(94.130.140.225:30304)), a block execution error: BlockImportFailed(BlockOmmersHashError)
12:32:45 [i.i.e.blockchain.sync.RegularSync] - No peers to download from`

Commit:
Add RegularSyncState case class and apply PR comments

Commit:
Add missing startSyncing call

Commit:
Fix failing tests and remove useless time value

Commit:
Fix scaladoc generation fail

Commit:
Fix regularSync stoping:
- remove assertion and use if instead with log debug
- add context become in handlingBlockBranches

Commit:
Add more context become and remove blacklistedPeers from RegularSyncState

Commit:
Remove the excess of context become calls

Commit:
Get rid of head of empty list exception

Commit:
Do code update after merge with phase/release2_0 branch

Commit:
Resolve conflict in handleResponseToRequest in ResponseReceived for bodies

Commit:
Merge branch 'phase/release2_0' into refactor/ec-320_regularsync

mmrozek reviewed (approved):
I found only a few small stylistic issues. Except that it looks good

KonradStaniec reviewed (commented):
Looks good but it would be great not to update ets tests on `phase/release2_0` branch.

Commit:
Apply changes
- make in ancientBlockHash condition simpler by using maxNewBlockHashAge > 0 instead od comparing blocks
- move notDownloading method to RegularSyncState
- fix log formatting
- make more specific scalastyle for handleBlockImport
- add description to importingBlocks param

Commit:
Merge branch 'phase/release2_0' into refactor/ec-320_regularsync

Commit:
Add missing tests for
- NewBlock message
- BlockBodies message

Commit:
Add missing tests for
- NewBlock message
- BlockBodies message

Commit:
Remove a typo that prevents compilation
